### PR TITLE
FS Sections

### PR DESCRIPTION
**Pull request by fszwedo**
Section endpoints, to test run ```npm run start dev```
and check the requests for
GET /api/sections
POST /api/sections
examplary body
{
    "name":"Typescript",
    "startDate": "2020-01-01",
    "endDate": "2021-01-01",
    "testDate": "2020-12-12",
    "description": "elo"
}

GET /api/sections/:id
PUT /api/sections/:id
DELETE /api/sections/:id

### CODE REVIEW

File: Api/Test/SectionService.test.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+class TestSectionRepository implements SectionRepository {
+    private sections: Array<SectionDBModel> = [];
+    model: any;
+
+    async getAll(){
+        return this.sections;
+    };
+
+    async getById(id: mongoose.Types.ObjectId){
+        return this.sections.find(project => project._id === id);
```
Comment (kjreczek) [inline Api/Test/SectionService.test.ts:18]: 😳 this is literally the only thing i found, should be section here not project 🙈

**Review by kjreczek — COMMENTED** [review]
only one little change to make 😃

File: Api/Test/SectionService.test.ts (hunk 1 above)
Comment (fszwedo) [reply Api/Test/SectionService.test.ts]: copying went wrong :D fixed, thanks for noticing

**Review by kjreczek — APPROVED** [review]

**Review by JustynaSmoliga — COMMENTED** [review]
Endpoints work fine. One small suggestion, apart from that all good :)

File: Api/Src/Models/Section.ts [hunk 2]
```diff
@@ -1,7 +1,57 @@
...
+        validate: [testDateValidator, 'Test date must be between start and end dates!']
+    },
+    tests: {
+        type: Array,
+    },
+    testDescription: {
+        type: String,
+    },
+    referenceProjectId: {
+        type: String
```
[the remark this answers is not in the document]
Comment (fszwedo) [reply Api/Src/Models/Section.ts]: done @JustynaSmoliga  :)

**Review by JustynaSmoliga — APPROVED** [review]

**Pull request merged**